gateways/js/src/fjage.js

```diff
@@ -132,14 +132,17 @@ export class AgentID {
     msg.index = Number.isInteger(index) ? index : -1;
     const rsp = await this.owner.request(msg, timeout);
     var ret = Array.isArray(params) ? new Array(params.length).fill(null) : null;
-    if (!rsp || rsp.perf != Performative.INFORM || !rsp.param) return ret;
+    if (!rsp || rsp.perf != Performative.INFORM || !rsp.param) {
+      if (this.owner._returnNullOnError) return ret;
+      else throw new Error(`Unable to set ${this.name}.${params} to ${values}`);
+    }
     if (Array.isArray(params)) {
       if (!rsp.values) rsp.values = {};
       if (rsp.param) rsp.values[rsp.param] = rsp.value;
       const rvals = Object.keys(rsp.values);
       return params.map( p => {
         let f = rvals.find(rv => rv.endsWith(p));
-        return f ? rsp.values[f] : null;
```

Comment:
Is there any practical difference between `null` and `undefined`? Just curious...

Reply:
I think the original language definition was : 

> `undefined` means a variable has been declared but has not yet been assigned a value
> `null` is an assignment value. It can be assigned to a variable as a representation of no value 

From : https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

So you're right,  in this scenario we should return `undefined`.